torch/onnx/verification.py

```diff
@@ -243,6 +246,219 @@ def compare_ort_pytorch_model_with_input(input_args, input_kwargs):
+
+                graph_diff_report.extend(source_printout)
+
+                break
+
+        return "\n".join(graph_diff_report)
+
+
+def _check_jit_model_diff(
+    model: Union[torch.nn.Module, torch.jit.ScriptModule],
+    test_input_sets: Sequence[Tuple[Tuple[Any, ...], Mapping[str, Any]]],
+    export_options: _experimental.ExportOptions,
+) -> str:
+    if len(test_input_sets) < 2:
+        raise ValueError("Need at least two set of test inputs to compare.")
+
+    training = export_options.training
+    verbose = export_options.verbose
+
+    with utils.exporter_context(model, training, verbose):
+        ref_jit_graph = None
+        for args, kwargs in test_input_sets:
+            export_inputs = _prepare_input_for_export(args, kwargs)
+            model = utils._pre_trace_quant_model(model, export_inputs)
+            jit_graph, _, _, _ = utils._create_jit_graph(model, export_inputs)
+
+            if ref_jit_graph is None:
+                ref_jit_graph = jit_graph
+                continue
+
+            graph_diff_report = _GraphDiff(ref_jit_graph, jit_graph).diff_report()
+            if graph_diff_report:
+                return graph_diff_report
+
+    return ""
+
+
+def _check_onnx_model_diff(
+    model: Union[torch.nn.Module, torch.jit.ScriptModule],
+    test_input_sets: Sequence[Tuple[Tuple[Any, ...], Mapping[str, Any]]],
+    export_options: _experimental.ExportOptions,
+) -> str:
+    if len(test_input_sets) < 2:
+        raise ValueError("Need at least two set of test inputs to compare.")
+
+    # TODO: refactor utils.py to remove duplicated code of context setup.
+    opset_version = export_options.opset_version
+    operator_export_type = export_options.operator_export_type
+    export_modules_as_functions = export_options.export_modules_as_functions
+    training = export_options.training
+    verbose = export_options.verbose
+    dynamic_axes = export_options.dynamic_axes
+    input_names = export_options.input_names
+    output_names = export_options.output_names
+
+    if opset_version is None:
+        opset_version = _constants.onnx_default_opset
+
+    export_modules_as_functions = utils._setup_trace_module_map(
+        model, export_modules_as_functions
+    )
+
+    if not operator_export_type:
+        if _C_onnx._CAFFE2_ATEN_FALLBACK:
+            operator_export_type = _C_onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK
+        else:
+            operator_export_type = _C_onnx.OperatorExportTypes.ONNX
+
+    _globals.GLOBALS.export_onnx_opset_version = opset_version
+    _globals.GLOBALS.operator_export_type = operator_export_type
+
+    with utils.exporter_context(model, training, verbose):
+        val_do_constant_folding = utils._decide_constant_folding(
+            export_options.do_constant_folding, operator_export_type, training
+        )
+
+        if dynamic_axes is None:
+            dynamic_axes = {}
+        utils._validate_dynamic_axes(dynamic_axes, model, input_names, output_names)
```

Comment:
I think it makes sense. The code is mimicking what `_export` is doing, so I'd avoid changing it now. I'd want to refactor(distill) the exporter context setup part to avoid this duplication.

Reply:
Sounds good